src/modules/public/blacklistAccess/sagas/blacklistAccessFetchSaga.ts

```diff
@@ -0,0 +1,43 @@
+import { call, put } from 'redux-saga/effects';
+import { API, RequestOptions } from '../../../../api';
+import { alertPush } from '../../alert';
+import { sendAccessTokenData, sendAccessTokenError, SendAccessTokenFetch } from '../actions';
+
+const requestOptions: RequestOptions = {
+    apiVersion: 'barong',
+};
+
+export function* blacklistAccessFetchSaga(action: SendAccessTokenFetch) {
+    try {
+        const response = yield call(API.post(requestOptions), '/identity/users/access', action.payload);
+        yield put(sendAccessTokenData());
+
+        if (response) {
+            const restricted = localStorage.getItem('restricted');
+            const underMaintenance = localStorage.getItem('maintenance');
+
+            if (restricted) {
+                localStorage.removeItem('restricted');
+            }
+
+            if (underMaintenance) {
+                localStorage.removeItem('maintenance');
+            }
+
+            window.location.replace('/trading');
```

Comment:
Remove this from here, location must not be handled in saga. You call action that triggers this saga only in one place ```MagicLinkWrapper``` screen, so handle redirects there and use history.push() instead ```window.location.replace```)()